.github/workflows/unit_tests.yml

```diff
@@ -12,41 +12,51 @@ jobs:
     runs-on: ${{ matrix.os }}
     strategy:
       matrix:
-        os: [ubuntu-latest, macos-latest]
-        python-version: [3.8, 3.9]
+        os: [ubuntu-22.04,macos-14]
+        python-version: [3.8,3.9]
 
     steps:
       - uses: actions/checkout@v4
-      - uses: conda-incubator/setup-miniconda@v3
-        with:
-          activate-environment: test
+      - name: Set up Python ${{ matrix.python-version }}
+        uses: actions/setup-python@v5
+        id: cp310
+        with:    
           python-version: ${{ matrix.python-version }}
-          fetch-depth: 2
       - name: Install ubuntu dependencies
+        if: contains(matrix.os, 'ubuntu')
         shell: bash -el {0}
         run: |
-          if [ "${{ matrix.os }}" == "ubuntu-latest" ]; then
             sudo add-apt-repository ppa:ubuntu-toolchain-r/test
             sudo apt-get update
             sudo apt-get install libopenmpi-dev openmpi-bin
-          fi;
+
+      - name: Install MacOs dependencies
+        if: contains(matrix.os, 'macos')
+        shell: bash -el {0}
+        run: |
+              brew install open-mpi
+              echo "MPI_LIB_NRN_PATH=/opt/homebrew/lib/libmpi.dylib" >> $GITHUB_ENV
```

Comment:
this is not what we recommend for the users though ...

Reply:
I'm open to any suggestions

Reply:
Conda does not work? It seems to work on my mac and 2 other macs we tried on today ...

Reply:
The Linux build in the CI is failing because conda is installing a version of neuron without MPI support. Removing conda solves this issue, but it affects the mac-os build. This PR helps giving support to both systems without conda.

Reply:
have you read the parallel backend guide:

https://jonescompneurolab.github.io/hnn-core/stable/parallel.html 

? You should use conda for mac but pip for linux

Reply:
I split the os build and test on different workflows